bubblewrap.c

```diff
@@ -2351,6 +2351,40 @@ main (int    argc,
   if (umount2 ("oldroot", MNT_DETACH))
     die_with_error ("unmount old root");
 
+  /* This is our second pivot. It's like we're a Silicon Valley startup flush
+   * with cash but short on ideas!
+   *
+   * We're aiming to make /newroot the real root, and get rid of /oldroot. To do
+   * that we need a temporary place to store it before we can unmount it.
+   */
+  { cleanup_free char *pivot_tmp = xstrdup ("bwrap-pivot-old-XXXXXX");
+    if (mount ("/newroot", "/newroot", NULL, MS_MGC_VAL | MS_BIND | MS_REC, NULL) < 0)
+      die_with_error ("setting up newroot bind");
+    if (chdir ("/newroot") != 0)
+      die_with_error ("chdir /newroot");
+    if (mkdtemp (pivot_tmp) == NULL)
```

Comment:
I don't like how this writes to the new root. For example, this will not really work if you do a `--ro-bind / /`, or if the root is a read-only container image. Testing in `tmp` too makes it a bit more likely that it would work, but there is no guarantees that there is a tmp, or that it is writable. I really would like us to have minimal requiements for the target filesystem layout.

Reply:
barely tested but would something like the fixup be enough?

Also, the original patch is coming from @cgwalters.  Does something like in the fixup look good to you?

Reply:
> For example, this will not really work if you do a --ro-bind / /, or if the root is a read-only container image. 

I use the  `--ro-bind / /` pattern a lot myself for quick sanity checking of bwrap, but it feels like we really want a `--bind-host` or something? 

Are there really use cases for spawning a container with no writable filesystems?

Actually, couldn't we special case this and detect the case where the user is binding `/` and skip doing the tmpfs etc. entirely?

Reply:
the last patch I've added drops the requirement for write access to the destination root:

```
./bwrap --ro-bind / / --proc /proc ./bwrap --ro-bind / / echo hello
hello
```